core/test/scripts/Voting.js

```diff
@@ -147,4 +147,47 @@ contract("scripts/Voting.js", function(accounts) {
     assert.equal(await voting.getPrice(identifier1, time2), hardcodedPrice);
     assert.equal(await voting.getPrice(identifier2, time2), hardcodedPrice);
   });
+
+  it.only("CryptoCompare price", async function() {
+    const identifier = web3.utils.utf8ToHex("BTCUSD");
+    const time = "1560762000";
+
+    // Request an Oracle price.
+    await voting.addSupportedIdentifier(identifier);
+    await voting.requestPrice(identifier, time);
+
+    // Move to the round in which voters will vote on the requested price.
+    await moveToNextRound(voting);
+
+    // Sanity check.
+    assert.isFalse(await voting.hasPrice(identifier, time));
+
+    const emailSender = new MockEmailSender();
+    let votingSystem = new VotingScript.VotingSystem(voting, encryptedSender, voter, emailSender);
+
+    assert.equal(emailSender.emailsSent, 0);
+    // The vote should have been committed.
+    await votingSystem.runIteration();
+    assert.equal(emailSender.emailsSent, 1);
+    // Running again shouldn't send more emails.
+    await votingSystem.runIteration();
+    assert.equal(emailSender.emailsSent, 1);
+
+    // Move to the reveal phase.
+    await moveToNextPhase(voting);
+
+    // Replace the voting system object with a new one so the class can't persist the commit.
+    votingSystem = new VotingScript.VotingSystem(voting, encryptedSender, voter, emailSender);
+
+    // This vote should have been removed from the persistence layer so we don't re-reveal.
```

Comment:
nit: I think we can remove this entire section since the iterations don't do anything - they just check to make sure the emails aren't sent again.